### PR TITLE
Harden the kernel package filter just-in-time compiler operation.

### LINKED ISSUE #4494: need new rule: bpf_jit_harden
#echo "net.core.bpf_jit_harden = 2" >> $CONFIG

### PR DESCRIPTION
**Pull request by adelton**
#### Description:

- Harden the kernel package filer just-in-time compiler operation.

#### Rationale:

- When hardened, the extended Berkeley Packet Filter just-in-time compiler will randomize any kernel addresses in the BFP programs and maps, and will not expose the JIT addresses in `/proc/kallsyms`.

Fixes #4494

### CODE REVIEW

File: linux_os/guide/system/permissions/restrictions/sysctl_net_core_bpf_jit_harden/rule.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+prodtype: rhel8,fedora,ol8
+
+title: 'Harden the operation of the BPF just-in-time compiler'
+
+description: '{{{ describe_sysctl_option_value(sysctl="net.core.bpf_jit_harden", value="2") }}}'
+
+rationale: |-
+    When hardened, the extended Berkeley Packet Filter just-in-time compiler
+    will randomize any kernel addresses in the BFP programs and maps,
```
Comment (jan-cerny) [inline linux_os/guide/system/permissions/restrictions/sysctl_net_core_bpf_jit_harden/rule.yml]: BFP should be BPF

File: linux_os/guide/system/permissions/restrictions/sysctl_net_core_bpf_jit_harden/rule.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    and will not expose the JIT addresses in <tt>/proc/kallsyms</tt>.
+
+severity: medium
+
+identifiers:
+    cce@rhel8: 82934-1
+
+references:
+    anssi: NT28(R25)
```
Comment (jan-cerny) [inline linux_os/guide/system/permissions/restrictions/sysctl_net_core_bpf_jit_harden/rule.yml]: ANSSI NT28 R25 is about YAMA, so the reference should be removed.

File: linux_os/guide/system/permissions/restrictions/sysctl_net_core_bpf_jit_harden/rule.yml (hunk 1 above)
Comment (adelton) [reply linux_os/guide/system/permissions/restrictions/sysctl_net_core_bpf_jit_harden/rule.yml]: Fixed in 5de079de06a58e6009f477ca19f4b86038f0634f, thanks for spotting it.

File: linux_os/guide/system/permissions/restrictions/sysctl_net_core_bpf_jit_harden/rule.yml (hunk 2 above)
Comment (adelton) [reply linux_os/guide/system/permissions/restrictions/sysctl_net_core_bpf_jit_harden/rule.yml]: Removed in 5de079de06a58e6009f477ca19f4b86038f0634f.

**Comment by adelton** [comment]
Addressed review, rebased on master -> 5de079de06a58e6009f477ca19f4b86038f0634f.

**Comment by adelton** [comment]
I will rebased on master once https://github.com/ComplianceAsCode/content/pull/4563 is merged. I have prepared fix of the typo in the commit message.

**Comment by adelton** [comment]
Rebased on master -> 182830f816d46cd8869202fde6f8a64c02e5057a.

**Comment by jan-cerny** [comment]
@adelton thank you

**Pull request merged**